lib/player.js

```diff
@@ -1133,69 +957,49 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
   }
 
   /**
-   * Tell the player to stop using its current media element. If the player is:
-   * <ul>
-   *  <li>detached, this will do nothing,
-   *  <li>attached, this will release the media element,
-   *  <li>loading, this will abort loading, unload, and release the media
-   *      element,
-   *  <li>playing content, this will stop playback, unload, and release the
-   *      media element.
-   * </ul>
+   * TODO: Desc.
    *
-   * <p>
-   * Calls to <code>detach</code> will interrupt any in-progress calls to
-   * <code>load</code> but cannot interrupt calls to <code>attach</code>,
-   * <code>detach</code>, or <code>unload</code>.
+   * @param {boolean=} keepAdManager
    *
    * @return {!Promise}
    * @export
    */
-  detach() {
+  async detach(keepAdManager = false) {
     // Do not allow the player to be used after |destroy| is called.
     if (this.loadMode_ == shaka.Player.LoadMode.DESTROYED) {
-      return Promise.reject(this.createAbortLoadError_());
+      throw this.createAbortLoadError_();
     }
 
-    // Tell the walker to go "detached", but do not allow it to be interrupted.
-    // If it could be interrupted it means that our media element could fall out
-    // of sync.
-    const events = this.walker_.startNewRoute((currentPayload) => {
-      return {
-        node: this.detachNode_,
-        payload: shaka.Player.createEmptyPayload_(),
-        interruptible: false,
-      };
-    });
+    await this.unload(/* initializeMediaSource= */ false);
```

Comment:
We need propagate the param keepAdManager to this function

Reply:
Done.